src/lib.rs

```diff
@@ -523,8 +535,27 @@ impl<'a, Writer: Write> Printer<'a, Writer> {
         Ok(())
     }
 
+    fn reorder_buf_to_le(&self, buf: &mut Vec<u8>) {
+        let n = buf.len();
+        let group_sz = self.group_size as usize;
+
+        for idx in (0..n).step_by(group_sz) {
+            let remaining = n - idx;
+            let total = remaining.min(group_sz);
+
+            buf[idx..idx + total].reverse();
+        }
+    }
+
     pub fn print_bytes(&mut self) -> io::Result<()> {
-        for (i, &b) in self.line_buf.clone().iter().enumerate() {
+        let mut buf = self.line_buf.clone();
+
+        if self.little_endian_dump {
+            // reorder the buffer to the little endian format
+            self.reorder_buf_to_le(&mut buf);
```

Comment:
The character panel is not affected by this reordering. Is this by design? `xxd` does the same, but I'm not entirely sure why?

Reply:
In my option, the endianness only makes sense for the "grouping" bytes representation. For the character panel, each character will directly correspond to one single byte, which has no difference between big or little endianness.

Reply:
Ok. I just thought it would be confusing as there is no 1:1 correspondence anymore when using little-endian mode. But let's leave things as they are for now.